src/snapshot/src/lib.rs

```diff
@@ -77,73 +77,27 @@ pub struct Snapshot {
 }
 
 #[derive(Default, Debug, Versionize)]
-pub struct Section {
+struct Section {
     name: String,
     data: Vec<u8>,
 }
 
-#[derive(PartialEq)]
-pub enum Error {
-    Io(i32),
-    Serialize(String),
-    Deserialize(String),
-    Semantic(String),
-    Crc64,
-}
-
-pub type Result<T> = std::result::Result<T, Error>;
-
-impl fmt::Display for Error {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            Error::Io(ref err) => write!(f, "IO error: {}", err),
-            Error::Serialize(ref err) => write!(f, "Serialization error: {}", err),
-            Error::Deserialize(ref err) => write!(f, "Deserialization error: {}", err),
-            Error::Semantic(ref err) => write!(f, "Semantic error: {}", err),
-            Error::Crc64 => write!(f, "Crc64 check failed"),
-        }
-    }
-}
-
-impl fmt::Debug for Error {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            Error::Io(ref err) => write!(f, "IO error: {}", err),
-            Error::Serialize(ref err) => write!(f, "Serialization error: {}", err),
-            Error::Deserialize(ref err) => write!(f, "Deserialization error: {}", err),
-            Error::Semantic(ref err) => write!(f, "Semantic error: {}", err),
-            Error::Crc64 => write!(f, "Crc64 check failed"),
-        }
+// Returns format version if arch id is valid.
+// Returns none otherwise.
+fn validate_magic_id(magic_id: u64) -> Result<u16> {
```

Comment:
nit: can we rename this function to `compute_format_version_from_magic_id` or `get_format_version_from` or something like this to better suggest its purpose?

Reply:
done.